bip-0360.mediawiki

```diff
@@ -511,18 +511,40 @@ attacks.
 
 ==== Transaction Size and Fees ====
 
-P2QRH inputs can be slightly larger or smaller than their equivalent P2TR inputs. Lets consider the case:
+P2QRH inputs can be slightly larger or smaller than their equivalent P2TR inputs. Let's consider the cases:
 
-'''P2TR Key-spend''' P2QRH inputs will be larger than P2TR inputs when the P2TR output would have been spent via the key-spend path.
-P2QRH quantum resistance comes from removing the P2TR key-spend path. Consequently it can not make use of taproot's optimization
-where P2TR key-spends do not require including a merkle path in the P2TR input. The size difference is roughly:
-len(tapleaf executed) + 32m - len(pubkey) - len(signature) bytes, where m=log2(number of tapleafs). For a single tapleaf 
-P2QRH would be 4 bytes longer.
+'''P2TR key-spend''' P2QRH inputs will be larger than P2TR inputs when the P2TR output would have been spent via the key-spend path.
+P2QRH quantum resistance comes from removing the P2TR key-spend path. Consequently it cannot make use of taproot's optimization
+where P2TR key-spends do not require including a merkle path in the P2TR input. If the Merkle tree only has a single tapleaf, 
+no Merkle path is needed in the control block giving use a 1 byte control block and the 34 byte tapleaf itself.
 
+P2QRH witness (99 bytes)
+<source>
+signature (64 bytes),
+tapleaf script = [OP_PUSHBYTES_32, 32 byte public key, OP_CHECKSIGVERIFY] (34 bytes),
+control block = [control byte]  (1 byte)
+</source>
```

Comment:
Should we also count the compact size used to indicate the length of each witness field?
https://learnmeabitcoin.com/technical/transaction/witness/#structure